app/models/user.rb

```diff
@@ -61,4 +61,96 @@ def generate_token(column)
       break unless User.exists?(column => self[column])
     end
   end
+
+  # @return [TrueClass, FalseClass]
+  def uploaded_before? = !user_resources.all.empty?
+
+  # noinspection RubyNilAnalysis
+  def last_uploaded_file
+    return 'none' unless uploaded_before?
+
+    user_resources.last.name
+  end
+
+  # @return Array[Hash]
+  def upload_file_history
+    upload_file_history = []
+    all_upload_dates.each do |item|
+      upload_file_history.append({ 'date': item, 'number': get_number_of_files_at_date(item) })
+    end
+    upload_file_history
```

Comment:
Warum nicht einfach `user_resources.group(:creation_date).count` und fertig ist die Kiste.

https://edgeguides.rubyonrails.org/active_record_querying.html#total-of-grouped-items